ravenml/train/interfaces.py

```diff
@@ -16,101 +21,142 @@ class TrainInput(object):
+            place their specific metadata. Plugins should never assign to this
+            attribute as it will break the relationship between plugin_metadata and metadata.
+        plugin_config (dict): plugin section of config dict. Plugins look here
+            for plugin-specific configuration.
     """
-    def __init__(self, inquire=True, **kwargs):
-        if inquire:
-            self._artifact_path = Path(os.path.expanduser(user_input('Enter filepath for artifacts:'))) if \
-                                    user_confirms('Run in local mode? (No S3 upload)') else None
-            dataset_options = cli_spinner('Finding datasets on S3', get_dataset_names)
-            dataset = user_selects('Choose dataset:', dataset_options)
-            self._dataset = cli_spinner(f'Downloading {dataset} from S3...', get_dataset, dataset)
-        else:
-            self._dataset = kwargs.get('dataset')
-            self._artifact_path = Path(kwargs.get('artifact_path'))
-    
-    def local_mode(self):
-        return self.artifact_path is not None
+    def __init__(self, config:dict=None, plugin_name:str=None):
+        """ Keyword args must be used for this class to work with the @pass_train pass decorator.
+        That decorator is defined in options.py. It is applied to commands to ensure that a 
+        TrainInput object exists in the click Context at ctx.obj. Click pass decorators 
+        do this by creating an object with no constructor arguments at ctx.obj if one does
+        not already exist.
 
-    @property
-    def dataset(self):
-        return self._dataset
-        
-    @dataset.setter
-    def dataset(self, new_dataset: Dataset):
-        self._dataset = new_dataset
+        Therefore, this constructor is ONLY called with no args when a command decorated with 
+        @pass_train is triggered and no TrainInput exists yet at ctx.obj. This behavior is used
+        to know when the user has called a plugin training command but did not provide a config 
+        filepath via --config.
         
-    @property
-    def artifact_path(self):
-        return self._artifact_path
+        The train command defined at commands.py will ONLY create a TrainInput at ctx.obj
+        if --config was used. This prevents undesired creation of TrainInput objects during
+        execution non-training plugin commands, as click always executes the code in the train
+        command on its way down the chain to the plugin command itself.
         
-    @artifact_path.setter
-    def artifact_path(self, new_path: str):
-        self._artifact_path = new_path
+        tl:dr: if this constructor is called without keyword args set, we know a user
+        has called a plugin's train command without providing a config, so we fail.
+        """
+        if config is None or plugin_name is None:
+            raise click.exceptions.UsageError(('You must provide the --config option'
```

Comment:
Minor typo needs an extra space, shows up as ` You must provide the --config optionon `ravenml train``

Reply:
good catch, fixed